### PR TITLE
Integration of Heppy in CMSSW_7_2_X.

### PR DESCRIPTION
**Pull request by cbernet**
The changes have been tested in CMSSW_7_2_2_patch2 and affect two packages only, PhysicsTools/Heppy and PhysicsTools/HeppyCore. These new packages have already been integrated in CMSSW_7_3_X. 

@vadler @gpetruc @arizzi

### CODE REVIEW

File: PhysicsTools/Heppy/interface/MuScleFitCorrector.h [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    double pt = lorentzVector.Pt();
+    double eta = lorentzVector.Eta();
+    double squaredDiff = getSigmaPtDiffSquared(pt,eta);
+    if (squaredDiff < 0) return pt;
+    double Cfact = 0.;
+    if (fileName_.Contains("smearPrompt")) Cfact = 0.85; // smear Summer12 against 2012 Prompt data
+    else if (fileName_.Contains("smearReReco")) Cfact = 0.75; // smear Summer12 against 2012 ReReco data
+    else if (fileName_.Contains("2011_MC")) Cfact = 0.8; // smear Fall11 against 2011 ReReco data
+    else {
+      std::cout<<"Are you sure you want to smear data??"<<std::endl;
```
Comment (degano) [inline PhysicsTools/Heppy/interface/MuScleFitCorrector.h]: Please do not use std::cout, edm::LogInfo or edm::LogWarning are the supported methods.

File: PhysicsTools/Heppy/interface/MuScleFitCorrector.h [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	}
+	
+	scaleFunctionId_ = scaleFunctionNum;
+	scaleFunction_ = scaleFunctionService(scaleFunctionNum);
+	resolutionFunctionId_ = resolutionFunctionNum;
+	resolutionFunction_ = resolutionFunctionService(resolutionFunctionNum);
+	
+	/*       std::cout<<"Function IDs: "<<std::endl; */
+	/*       std::cout<<"     scale function number "<<scaleFunctionId_<<std::endl; */
+	/*       std::cout<<"     resolution function number "<<resolutionFunctionId_<<std::endl; */
```
Comment (degano) [inline PhysicsTools/Heppy/interface/MuScleFitCorrector.h]: Please remove commented code.

File: PhysicsTools/Heppy/interface/MuScleFitCorrector.h (hunk 1 above)
Comment (arizzi) [inline PhysicsTools/Heppy/interface/MuScleFitCorrector.h]: this is not running in full framework, using LogInfo/LogWarning makes no sense

File: PhysicsTools/Heppy/interface/MuScleFitCorrector.h (hunk 2 above)
Comment (arizzi) [inline PhysicsTools/Heppy/interface/MuScleFitCorrector.h]: why should people remove commented code that could be useful for debugging??? 
comments, including stuff that can help debugging should be welcome unless they completly clutter the rest of the code (not the case here)

**Comment by degano** [comment]
@cmsbuild please test

File: PhysicsTools/Heppy/interface/MuScleFitCorrector.h (hunk 2 above)
Comment (cbernet) [inline PhysicsTools/Heppy/interface/MuScleFitCorrector.h]: I support both comments from @arizzi, these requests don’t make much sense. 

Cheers, 

Colin

**Comment by monttj** [comment]
+1 
These packages have been already integrated in 73X and 74X.

**Comment by davidlange6** [comment]
+1

**Pull request merged**